src/rebuild/srv.c

```diff
@@ -338,26 +450,30 @@ rebuild_global_status_update(struct rebuild_global_pool_tracker *rgt,
 	if (!is_rebuild_global_scan_done(rgt)) {
 		rebuild_leader_set_status(rgt, iv->riv_rank, iv->riv_dtx_resyc_version,
 					  SCAN_DONE);
-		D_DEBUG(DB_REBUILD, "rebuild ver %d tgt %d scan done\n",
-			rgt->rgt_rebuild_ver, iv->riv_rank);
+		D_DEBUG(DB_REBUILD, DF_RB ": rank %d scan done\n", DP_RB_RGT(rgt), iv->riv_rank);
 		/* If global scan is not done, then you can not trust
 		 * pull status. But if the rebuild on that target is
 		 * failed(riv_status != 0), then the target will report
 		 * both scan and pull status to the leader, i.e. they
 		 * both can be trusted.
 		 */
-		if (iv->riv_status == 0)
+		if (iv->riv_status == 0) {
+			/* test only: update_and_warn_for_slow_engines(rgt); */
 			return 0;
+		}
 	}
 
-	/* Only trust pull done if scan is done globally */
+	/* Only trust pull done if scan errored or is done globally */
 	if (iv->riv_pull_done) {
-		rebuild_leader_set_status(rgt, iv->riv_rank, iv->riv_dtx_resyc_version,
-					  PULL_DONE);
-		D_DEBUG(DB_REBUILD, "rebuild ver %d tgt %d pull done\n",
-			rgt->rgt_rebuild_ver, iv->riv_rank);
+		rebuild_leader_set_status(rgt, iv->riv_rank, iv->riv_dtx_resyc_version, PULL_DONE);
+		D_DEBUG(DB_REBUILD, DF_RB ": rank %d pull done\n", DP_RB_RGT(rgt), iv->riv_rank);
+		if (iv->riv_status != 0)
+			DL_WARN(iv->riv_status, DF_RB ": rank %u update with failure",
```

Comment:
maybe can define a fixed format sub-string like (DF_RB" rebuild_status_update_leader: ") on leader and (DF_RB" rebuild_status_update_target: ")  to all logs that may affect the rebuild status (for all cases like DONE, WARN, fail report etc) to facilitate the log parsing. FYI.

Reply:
OK let me consider this some more in future updates. To be clear, you mean that relying on the function name that is emitting the log is not as good a way to differentiate between target-engine activities versus leader-engine activities for status reporting? e.g., rebuild_tgt_status_check_ult() to represent all target engines (I guess including the leader)? and rebuild_global_status_update() to represent the leader processing the updates from all engines?